core/src/main/java/org/apache/accumulo/core/client/ClientSideIteratorScanner.java

```diff
@@ -297,4 +345,50 @@ public void setReadaheadThreshold(long batches) {
     }
     this.readaheadThreshold = batches;
   }
+
+  private SamplerConfiguration getIteratorSamplerConfigurationInternal() {
+    SamplerConfiguration scannerSamplerConfig = getSamplerConfiguration();
+    if (scannerSamplerConfig != null) {
+      if (iteratorSamplerConfig != null && !new SamplerConfigurationImpl(iteratorSamplerConfig).equals(new SamplerConfigurationImpl(scannerSamplerConfig))) {
```

Comment:
fixed in f6f60ad .   At one point `SamplerConfiguration` did not have `equals()` impl, but now it does so I shortened it.